lib/notify.c

```diff
@@ -119,3 +120,22 @@ system_call_script(thread_master_t *m, int (*func) (thread_t *), void * arg, lon
 
 	exit(WEXITSTATUS(status));
 }
+
+void
+script_killall(thread_master_t *m, int signo)
+{
+	thread_t *thread;
+	pid_t p_pgid, c_pgid;
+
+	thread = m->child.head;
+
+	p_pgid = getpgid(0);
+
+	while (thread) {
```

Comment:
I think this loop needs to be called with SIGCHLD blocked, otherwise the `thread_child_handler()` signal handler may run and modify the child queue while we are modifying it.